Sources/ScreenObject/ScreenObject.swift

```diff
@@ -73,13 +65,15 @@ open class ScreenObject {
         self.app = app
         self.expectedElementGetters = [expectedElementGetter]
         self.waitTimeout = waitTimeout
-        try waitForFirstScreenElement()
+        try waitForScreen(firstElementOnly: true)
     }
 
     @discardableResult
-    func waitForScreen() throws -> Self {
+    public func waitForScreen(firstElementOnly: Bool = false) throws -> Self {
+        let testedGetters = firstElementOnly ? [expectedElementGetters.first!] : expectedElementGetters
```

Comment:
Force unwrapping the first element has the (low) chance to crash the tests. That's not a big deal because a crash is still reported as a failure, but has the disadvantage of stopping the test run which might result in losing some information from the tests that don't get a chance to run.

Have you considered throwing an error instead?

All the infrastructure to do it is already in place, including a dedicated error:

```swift
guard let firstGetter = expectedElementGetters.first else {
    throw InitError.emptyExpectedElementGettersArray
}

let testedGetters = firstElementOnly ? firstGetter : expectedElementGetters
```

If we throw that error, the test that called the method with an empty ScrenObject will fail in a clearer way.